src/obj_char_transfer.c

```diff
@@ -0,0 +1,872 @@
+#include "global.h"
+#include "obj_char_transfer.h"
+#include "vram_transfer_manager.h"
+
+typedef enum ObjCharTransferTaskState {
+    OBJ_CHAR_TRANSFER_TASK_INIT,
+    OBJ_CHAR_TRANSFER_TASK_1,
+    OBJ_CHAR_TRANSFER_TASK_2,
+    OBJ_CHAR_TRANSFER_TASK_COPYABLE,
+    OBJ_CHAR_TRANSFER_TASK_COPY,
+} ObjCharTransferTaskState;
+
+typedef struct ObjCharTransferTask {
+    NNSG2dCharacterData *charData;
+    NNS_G2D_VRAM_TYPE vram;
+    u8 isTransferred;
+    int resId;
+    NNSG2dImageProxy imageProxy;
+    u32 baseAddrMain;
+    u32 baseAddrSub;
+    u8 state;
+    BOOL shouldUpdateMappingTypeFromHW;
+    BOOL transferFromTail;
+    BOOL needResetBlockTransferFlags;
+    u32 sizeMain;
+    u32 sizeSub;
+} ObjCharTransferTask;
+
+struct ObjCharTransferTasksManager {
+    ObjCharTransferTask *tasks;
+    int max;
+    int num;
+    int offsetMain;
+    int offsetSub;
+    u32 vramCapacityMain;
+    u32 vramCapacitySub;
+    int freeSizeMain;
+    int freeSizeSub;
+    int numBlocksMain;
+    int numBlocksSub;
+    u32 blockSizeMain;
+    u32 blockSizeSub;
+    u8 *blockBufMain;
+    u8 *blockBufSub;
+} *sObjCharTransferTasksManager;
+
+static BOOL ObjCharTransfer_TaskExistsByID(int resId);
+static void resetAllTransferTasks(void);
```

Comment:
should really have a prefix

Reply:
still leaning towards prefix for this, I can understand why you may not want to have it though